lib/board.rb

```diff
@@ -15,4 +15,96 @@ def create_cells
             end
         end
     end
-end
\ No newline at end of file
+
+
+
+  
+#     def initialize
+#       @cells = create_cells
+#     end
+  
+#     def create_cells
+#       cells = {}
+#       all_coordinates.each do |coordinate|
+#         cells[coordinate] = Cell.new(coordinate)
+#       end
+#       cells
+#     end
+
+  
+#     def all_coordinates
+#       coordinates = []
+#       rows.each do |row|
+#         columns.each do |column|
+#           coordinates << "#{row}#{column}"
+#         end
+#       end
+#       coordinates
+#     end
+  
+
+#     def rows
+#       ["A", "B", "C", "D"]
+#     end
+  
+
+#     def columns
+#       [1, 2, 3, 4]
+#     end
+#   end
+
+
+def valid_coordinate?(coordinate)
+    @cells.key?(coordinate)
+end
```

Comment:
Nice! The interaction pattern has a few additional tests for this method, so suggest we add those tests as well(below). They all passed when I tried them. 

it 'can make sure coordinates are valid' do 

            expect(@board.valid_coordinate?("A1")).to eq(true)
            expect(@board.valid_coordinate?("D4")).to eq(true)
            expect(@board.valid_coordinate?("A5")).to eq(false)
            expect(@board.valid_coordinate?("E1")).to eq(false)
            expect(@board.valid_coordinate?("A22")).to eq(false)
        end